### PR TITLE
compose.go changes

### PR DESCRIPTION
**Pull request by kaushik691**
Modified compose.go file to accept alias and custom alpha flags:
1. `--alias` - alias for the containers
2. `--custom_alpha_flags` Custom alpha flags for specific alphas, following {"1:custom_flags", "2:custom_flags"},
eg: {"2: -p <bulk_path>"}
`go run compose.go -a=2 -z=1 --alias=test --custom_alpha_flags={"2:-p <bulk_path>"}`

---
This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/7872)

### CODE REVIEW

File: compose/compose.go [hunk 1]
```diff
@@ -725,6 +743,10 @@ func main() {
 		"minio service port")
 	cmd.PersistentFlags().StringArrayVar(&opts.MinioEnvFile, "minio_env_file", nil,
 		"minio service env_file")
+	cmd.PersistentFlags().StringVar(&opts.ContainerPrefix, "alias", "",
+		"alias for the containers")
```
Comment (NamanJain8) [inline compose/compose.go:747]:
```suggestion
		"prefix for the container name")
```

File: compose/compose.go [hunk 2]
```diff
@@ -725,6 +743,10 @@ func main() {
...
+	cmd.PersistentFlags().StringArrayVar(&opts.CustomAlphaOptions, "custom_alpha_flags", nil,
```
Comment (NamanJain8) [inline compose/compose.go:748]:
```suggestion
	cmd.PersistentFlags().StringArrayVar(&opts.CustomAlphaOptions, "custom_alpha_options", nil,
```

File: compose/compose.go [hunk 3]
```diff
@@ -725,6 +743,10 @@ func main() {
...
+		"Custom alpha flags for specific alphas, following {\"1:custom_flags\", \"2:custom_flags\"}, eg: {\"2: -p <bulk_path>")
```
Comment (NamanJain8) [inline compose/compose.go:749]: A line should not contain more than 100 chars. Split it into 2 lines.

File: compose/compose.go [hunk 4]
```diff
@@ -773,10 +795,24 @@ func main() {
 		services[svc.name] = svc
 	}
 
+	// Alpha Customization
+	customAlphas := make(map[int]string)
+	for _, flag := range opts.CustomAlphaOptions {
+		splits := strings.SplitN(flag, ":", 2)
+		if len(splits) != 2 {
+			fatal(errors.Errorf(" --custom_alpha_flags option requires string in index:options format."))
```
Comment (NamanJain8) [inline compose/compose.go:803]: `--custom_alpha_options`

File: compose/compose.go [hunk 5]
```diff
@@ -773,10 +795,24 @@ func main() {
...
+		}
+		custIdx, err := strconv.Atoi(splits[0])
+		if err != nil {
+			fatal(errors.Errorf(" --custom_alpha_flags captured erros while parsing index value %v", err))
```
Comment (NamanJain8) [inline compose/compose.go:807]: rename the flag?

File: compose/compose.go [hunk 6]
```diff
@@ -773,10 +795,24 @@ func main() {
...
 	}
 
+	// Alpha Customization
+	customAlphas := make(map[int]string)
+	for _, flag := range opts.CustomAlphaOptions {
+		splits := strings.SplitN(flag, ":", 2)
+		if len(splits) != 2 {
+			fatal(errors.Errorf(" --custom_alpha_flags option requires string in index:options format."))
+		}
+		custIdx, err := strconv.Atoi(splits[0])
```
Comment (NamanJain8) [inline compose/compose.go:805]: Rename it to `idx`.

File: compose/compose.go [hunk 7]
```diff
@@ -307,7 +316,12 @@ func getAlpha(idx int, raft string) service {
...
-	zeroHostAddr := fmt.Sprintf("%s:%d", getHost("zero1"), zeroBasePort+opts.PortOffset)
+	zeroName := "zero"
+	if opts.ContainerPrefix != "" {
+		zeroName = opts.ContainerPrefix + "_" + zeroName
+	}
+
+	zeroHostAddr := fmt.Sprintf("%s%d:%d", zeroName, 1, zeroBasePort+opts.PortOffset)
 	zeros := []string{zeroHostAddr}
 	for i := 2; i <= maxZeros; i++ {
 		port := zeroBasePort + opts.PortOffset + getOffset(i)
```
Comment (NamanJain8) [inline compose/compose.go:327]: You would need the containerPrefix logic in this for loop as well, I think.

File: compose/compose.go [hunk 8]
```diff
@@ -307,7 +316,12 @@ func getAlpha(idx int, raft string) service {
 		maxZeros = opts.NumZeros
 	}
 
-	zeroHostAddr := fmt.Sprintf("%s:%d", getHost("zero1"), zeroBasePort+opts.PortOffset)
+	zeroName := "zero"
+	if opts.ContainerPrefix != "" {
```
Comment (NamanJain8) [inline compose/compose.go:320]: `getHost()` logic somehow got removed? Don't we need that?

File: compose/compose.go (hunk 5 above)
Comment (kaushik691) [reply compose/compose.go:807]: no isssue we can

File: compose/compose.go (hunk 7 above)
Comment (kaushik691) [reply compose/compose.go:327]: done

File: compose/compose.go (hunk 8 above)
Comment (kaushik691) [reply compose/compose.go:320]: updated

**Review by NamanJain8 — APPROVED** [review]
Nice change @kaushik691. 🎉

**Pull request merged**